contracts/token/ERC721/ERC721.sol

```diff
@@ -441,52 +447,20 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
                     }
                 }
             }
-        } else {
-            return true;
         }
     }
 
-    /**
-     * @dev Hook that is called before any token transfer. This includes minting and burning. If {ERC721Consecutive} is
-     * used, the hook may be called as part of a consecutive (batch) mint, as indicated by `batchSize` greater than 1.
-     *
-     * Calling conditions:
-     *
-     * - When `from` and `to` are both non-zero, ``from``'s tokens will be transferred to `to`.
-     * - When `from` is zero, the tokens will be minted for `to`.
-     * - When `to` is zero, ``from``'s tokens will be burned.
-     * - `from` and `to` are never both zero.
-     * - `batchSize` is non-zero.
-     *
-     * To learn more about hooks, head to xref:ROOT:extending-contracts.adoc#using-hooks[Using Hooks].
-     */
-    function _beforeTokenTransfer(address from, address to, uint256 firstTokenId, uint256 batchSize) internal virtual {}
-
-    /**
-     * @dev Hook that is called after any token transfer. This includes minting and burning. If {ERC721Consecutive} is
-     * used, the hook may be called as part of a consecutive (batch) mint, as indicated by `batchSize` greater than 1.
-     *
-     * Calling conditions:
-     *
-     * - When `from` and `to` are both non-zero, ``from``'s tokens were transferred to `to`.
-     * - When `from` is zero, the tokens were minted for `to`.
-     * - When `to` is zero, ``from``'s tokens were burned.
-     * - `from` and `to` are never both zero.
-     * - `batchSize` is non-zero.
-     *
-     * To learn more about hooks, head to xref:ROOT:extending-contracts.adoc#using-hooks[Using Hooks].
-     */
-    function _afterTokenTransfer(address from, address to, uint256 firstTokenId, uint256 batchSize) internal virtual {}
-
     /**
      * @dev Unsafe write access to the balances, used by extensions that "mint" tokens using an {ownerOf} override.
      *
-     * WARNING: Anyone calling this MUST ensure that the balances remain consistent with the ownership. The invariant
-     * being that for any address `a` the value returned by `balanceOf(a)` must be equal to the number of tokens such
-     * that `ownerOf(tokenId)` is `a`.
+     * NOTE: the value is limited to type(uint128).max. This protect against _balance overflow. It is unrealistic that
+     * a uint256 would ever overflow from increments when these increments are bounded to uint128 values.
+     *
+     * WARNING: Increassing an account's balance using this function should go in pair with an override of the
+     * {_ownerOf} function that resolve the ownership of the corresponding tokens so that balances and ownerships
+     * remain consistent with one another.
```

Comment:
> This is generally true but in the case of ERC721Votes it's not.

I wouldn't say this. ERC721Votes is not invoking `_increaseBalance`, it's just overriding it to react to when it's invoked by another module.

That said, "should go" or "usually goes" both work for me.

Reply:
I see, and agree with the thinking. My concern is that it's explained that any `ownerOf` override should go along with an `_increaseBalance` but an `_increaseBalance` on its own might not be obvious. And because this is the documentation of `_increaseBalance`, the NatSpec should suggest what to consider when overriding.

My concrete suggestion is now to add an `IMPORTANT` block in the NatSpec of `ownerOf`:

```solidity
/**
 * @dev See {IERC721-ownerOf}.
 *
 * IMPORTANT: Any override that resolves the ownership by grouping sequential `tokenId`s (i.e. batching)
 * should be in pair with an override of the {_increaseBalance} function to keep balances consistent with
 * the {balanceOf} internal accounting accounting,
 */
```

And change this one to the following `IMPORTANT` block:

```suggestion
     * IMPORTANT: The {_update} function doesn't internally depend on this function as it updates the balances
     * by a fixed value of `1`. When implementing any mechanism that is triggered on `_update` invocations,
     * consider also overriding this function to support nonstandard balance updates.
```

Reply:
I added a note inspired by your last suggestion in `_update` which I think is the right place...